### PR TITLE
Ignore missing servers or snippets when removing from map

### LINKED ISSUE #274: [BUG] Run fails if nginx.server.snippets is not defined in pillar
If nginx.server.snippets is not defined in my pillar file, the run fails. Just adding ```{}``` as value is enough to work around the problem.

The issue seems to be occuring at the pop in nginx/snippets.sls.

Currently using https://github.com/saltstack-formulas/nginx-formula/commit/19203409aeb40b042cffb0e4d39bc66303f17842

### PR DESCRIPTION
**Pull request by anderbubble**
nginx.servers_config wants a lightened copy of the nginx map
to render as json; but, when it was trying to remove the
servers and snippets keys from the map it assumed their presence,
causing a KeyError if they were not present by its use of .pop().

While wrapping these in an "if" clause would likely be more
correct, along with replacing .pop() with del (if jinja even
supports that) the simplest change here is to just specify a
default value for .pop(), which obviates the KeyError.

Fixes #274

### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: nginx/servers_config.sls [hunk 1]
```diff
@@ -10,8 +10,8 @@
 {% set server_states = [] %}
 {#- _nginx is a lightened copy of nginx map intended to passed in templates #}
 {%- set _nginx = nginx.copy() %}
-{%- do _nginx.pop('snippets') %}
-{%- do _nginx.pop('servers') %}
+{%- do _nginx.pop('snippets', None) %}
+{%- do _nginx.pop('servers', None) %}
```
Comment (javierbertoli) [inline nginx/servers_config.sls:14]: Perhaps, following your comments, you can change this to
```jinja
{%- do _nginx.pop('snippets', None) if nginx.snippets is defined %}
{%- do _nginx.pop('servers', None) if nginx.servers is defined %}
```
wdyt?

**Review by javierbertoli — CHANGES_REQUESTED** [review]
@anderbubble good catch.

If possible, mind you do a few more changes that are relevant?

1. Add a similar fix in the [snippets.sls](https://github.com/saltstack-formulas/nginx-formula/blob/master/nginx/snippets.sls#L11-L12) file
2. Delete [this line](https://github.com/saltstack-formulas/nginx-formula/blob/master/test/salt/passenger/pillar/nginx.sls#L5), [this block](https://github.com/saltstack-formulas/nginx-formula/blob/master/test/salt/passenger/pillar/nginx.sls#L24-L27) and [these lines](https://github.com/saltstack-formulas/nginx-formula/blob/master/test/salt/passenger/pillar/nginx.sls#L67-L68) so we have a test for this fix.
3. Amend the commit message to something like `fix(config): ignore missing servers or snippets` so our automated versioning does not complain (check [here](https://github.com/saltstack-formulas/.github/blob/master/CONTRIBUTING.rst#relationship-between-commit-type-and-version-bump) for more details)

With these other minor changes, we'll have testing in place for the issue and it will be OK to be merged

**Comment by javierbertoli** [comment]
@anderbubble ping?

**Comment by anderbubble** [comment]
I'm here; just busy. I expect I can perform the additional changes you suggested; it just might take me a bit to get to it.

**Comment by javierbertoli** [comment]
@anderbubble wonderful! I'll wait for your changes to merge this.

**Comment by myii** [comment]
>     1. Add a similar fix in the [snippets.sls](https://github.com/saltstack-formulas/nginx-formula/blob/master/nginx/snippets.sls#L11-L12) file
> 
>     3. Amend the commit message to something like `fix(config): ignore missing servers or snippets` so our automated versioning does not complain (check [here](https://github.com/saltstack-formulas/.github/blob/master/CONTRIBUTING.rst#relationship-between-commit-type-and-version-bump) for more details)

Updated this PR to complete the above steps requested by @javierbertoli.  Includes resolving this review comment:

* https://github.com/saltstack-formulas/nginx-formula/pull/275#discussion_r612749949

**Review by javierbertoli — APPROVED** [review]

**Comment by javierbertoli** [comment]
Thanks you @anderbubble & @myii for this!

**Pull request merged**